packages/fxa-payments-server/server/lib/routes/navigation-timing.js

```diff
@@ -0,0 +1,76 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
+const joi = require('joi');
+const URL = require('url').URL;
+
+const TIMESTAMP_MS = joi
+  .number()
+  .integer()
+  .required();
+// Validate the subset of PerformanceNavgationTiming properties used here.
+// (Ref: https://www.w3.org/TR/navigation-timing-2/#dom-performancenavigationtiming)
+const navigationTimingSchema = joi.object().keys({
+  domainLookupStart: TIMESTAMP_MS.min(0),
+  domComplete: TIMESTAMP_MS.min(joi.ref('domInteractive')),
+  domInteractive: TIMESTAMP_MS.min(joi.ref('responseEnd')),
+  loadEventEnd: TIMESTAMP_MS.min(joi.ref('loadEventStart')),
+  loadEventStart: TIMESTAMP_MS.min(joi.ref('domComplete')),
+  name: joi
+    .string()
+    .uri()
+    .required(),
+  redirectStart: TIMESTAMP_MS,
+  requestStart: TIMESTAMP_MS.min(joi.ref('domainLookupStart')),
+  responseEnd: TIMESTAMP_MS.min(joi.ref('responseStart')),
+  responseStart: TIMESTAMP_MS.min(joi.ref('requestStart')),
+});
+
+module.exports = (geolocate, UAParser, statsd) => ({
```

Comment:
how come `UAParser` is injected vs `require`ed here?

Reply:
It makes testing a little easier, kind of.  Although `jest.mock` is pretty easy.  So, no real good reason. 😅